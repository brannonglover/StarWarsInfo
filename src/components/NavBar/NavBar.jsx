import React from 'react';
import { NavBarButtonStyle, NavBarStyle } from './NavBar.styled';

const NavBar = ({ setPage }) => {
  return (
    <NavBarStyle>
      <NavBarButtonStyle onClick={() => setPage("planets")}>Planets</NavBarButtonStyle>
      <NavBarButtonStyle onClick={() => setPage("people")}>People</NavBarButtonStyle>
    </NavBarStyle>
  )
}

export default NavBar;