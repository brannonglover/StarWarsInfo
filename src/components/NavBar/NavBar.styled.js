import styled from "styled-components";

export const NavBarStyle = styled.nav`
  display: grid;
  grid-template-columns: 15% 15%;
  justify-content: center;
`;