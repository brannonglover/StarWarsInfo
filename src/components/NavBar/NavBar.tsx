import React from "react";
import { NavBarStyle } from "./NavBar.styled";
import strings from "../../common/strings";
import { INavBarProps } from "../../common/types";

const NavBar: React.FC<INavBarProps> = (props: INavBarProps) => {
  const { setPage } = props;
  return (
    <NavBarStyle>
      <button onClick={() => setPage("planets")}>{strings.PLANETS}</button>
      <button onClick={() => setPage("people")}>{strings.PEOPLE}</button>
    </NavBarStyle>
  );
};

export default NavBar;
