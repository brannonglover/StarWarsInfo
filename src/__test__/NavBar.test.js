import React from 'react';
import { cleanup, render } from '@testing-library/react';
import NavBar from '../components/NavBar/NavBar';

afterEach(cleanup);

test('Display navbar component', () => {
  const { asFragment } = render(<NavBar />);

  expect(asFragment()).toMatchSnapshot();
});