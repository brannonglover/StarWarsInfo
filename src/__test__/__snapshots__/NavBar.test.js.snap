// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display navbar component 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdfBwQ jkLVzr"
  >
    <button>
      Planets
    </button>
    <button>
      People
    </button>
  </nav>
</DocumentFragment>
`;
